google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java

```diff
@@ -0,0 +1,543 @@
+   * @return a FeedDetails object containing relevant information.
+   */
+  private FeedDetails getFeed(
+      GoogleAdsClient googleAdsClient, long customerId, String feedResourceName) {
+    // Construct the query.
+    String query =
+        "SELECT feed.attributes FROM feed WHERE feed.resource_name = '" + feedResourceName + "'";
+
+    // Construct the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(String.valueOf(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Issue the search request.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Get the first result because we only need the single feed item we created previously.
+      GoogleAdsRow googleAdsRow = searchPagedResponse.getPage().getResponse().getResults(0);
+      // Get the attributes list from the feed and create a map with keys of each attribute and
+      // values of each corresponding ID.
+      List<FeedAttribute> feedAttributeList = googleAdsRow.getFeed().getAttributesList();
+      Map<String, Long> feedAttributeMap = new Hashtable<>();
+      for (FeedAttribute feedAttribute : feedAttributeList) {
+        feedAttributeMap.put(feedAttribute.getName().getValue(), feedAttribute.getId().getValue());
+      }
+      // Finally, construct a FeedDetails object so that we have a nice small
+      // wrapper around the details we care about for the next phase.
+      FeedDetails feedDetails =
+          new FeedDetails(
+              googleAdsRow.getFeed().getResourceName(),
+              feedAttributeMap.get("Page URL"),
+              feedAttributeMap.get("Label"));
+
+      return feedDetails;
+    }
+  }
+
+  /**
+   * Creates a field mapping for a given feed.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param feedDetails the relevant information about the feed and its attributes.
+   */
+  private void createFeedMapping(
+      GoogleAdsClient googleAdsClient, long customerId, FeedDetails feedDetails) {
+    // Map the FeedAttributeIds to the fieldId constants.
+    AttributeFieldMapping urlFieldMapping =
+        AttributeFieldMapping.newBuilder()
+            .setFeedAttributeId(Int64Value.of(feedDetails.getUrlAttributeId()))
+            .setDsaPageFeedField(DsaPageFeedCriterionField.PAGE_URL)
+            .build();
+    AttributeFieldMapping labelFieldMapping =
+        AttributeFieldMapping.newBuilder()
+            .setFeedAttributeId(Int64Value.of(feedDetails.getLabelAttributeId()))
+            .setDsaPageFeedField(DsaPageFeedCriterionField.LABEL)
+            .build();
+
+    // Map the FeedAttributeIds to the fieldId constants.
```

Comment:
Please update the comment. Looks like this was inadvertently copied from line 287.

Reply:
Done.